crates/libgraphql-core/src/test/golden_tests/test_runner.rs

```diff
@@ -0,0 +1,632 @@
+        };
+    }
+
+    builder.build().ok()
+}
+
+/// Test valid operations against a schema
+fn test_valid_operations(test_case: &GoldenTestCase, schema: &Schema) -> Vec<GoldenTestResult> {
+    let mut results = Vec::new();
+
+    if test_case.valid_operations.is_empty() {
+        return results;
+    }
+
+    // Test each valid operation with its own fragment registry
+    for op_test in &test_case.valid_operations {
+        let test_name = format!(
+            "{}/valid_operations/{}",
+            test_case.name,
+            op_test.path.file_name().unwrap().to_str().unwrap()
+        );
+
+        // Build fragment registry from just this operation file
+        let fragment_registry = match build_fragment_registry(schema, &[&op_test.path]) {
+            Ok(reg) => reg,
+            Err(err) => {
+                results.push(GoldenTestResult {
+                    test_name,
+                    passed: false,
+                    error_message: Some(format!("Failed to build fragment registry: {err}")),
+                    file_path: op_test.path.clone(),
+                    file_snippet: None,
+                });
+                continue;
+            }
+        };
+
+        let exec_doc_result =
+            ExecutableDocumentBuilder::from_file(schema, &fragment_registry, &op_test.path);
+
+        match exec_doc_result {
+            Ok(_) => {
+                results.push(GoldenTestResult {
+                    test_name,
+                    passed: true,
+                    error_message: None,
+                    file_path: op_test.path.clone(),
+                    file_snippet: None,
+                });
+            }
+            Err(errors) => {
+                let error_str = format!("{errors:?}");
+                let snippet = extract_snippet_with_error_marker(&op_test.path, 5).ok();
+
+                results.push(GoldenTestResult {
+                    test_name,
+                    passed: false,
+                    error_message: Some(format!("Expected: Valid operation\nGot: {error_str}")),
+                    file_path: op_test.path.clone(),
+                    file_snippet: snippet,
+                });
+            }
+        }
+    }
+
+    results
+}
+
+/// Test invalid operations against a schema
+fn test_invalid_operations(
+    test_case: &GoldenTestCase,
+    schema: &Schema,
+) -> Vec<GoldenTestResult> {
+    let mut results = Vec::new();
+
+    if test_case.invalid_operations.is_empty() {
+        return results;
+    }
+
+    // Test each invalid operation with its own fragment registry
+    for op_test in &test_case.invalid_operations {
+        let test_name = format!(
+            "{}/invalid_operations/{}",
+            test_case.name,
+            op_test.path.file_name().unwrap().to_str().unwrap()
+        );
+
+        // Try to build fragment registry from this operation file
+        let fragment_registry = match build_fragment_registry(schema, &[&op_test.path]) {
```

Comment:
Same here -- build a single FragmentRegistry per test suite (not one per operation test_case).